### DISCUSSION TITLE
Weather script

### DISCUSSION THREAD

**Discussion by Mathew-D** [Manjaro-Sway/manjaro-sway#877] [Show and tell]
I know currently we get the weather from https://manjaro-sway.download/weather/ which gets it from somewhere else but I was looking at other options and decided to make a free account with www.weatherapi.com which gives me 1mil updates a month.  To that end I created a script that gets the data from their api and converts it to a format I can use for waybar and wanted to post it in case someone else is interested. My version as my location hard coded and API hard coded but the script could be edited to take that as input.  The location can be given in many different ways as listed in the script. 

[weather_sh.txt](https://github.com/user-attachments/files/17895643/weather_sh.txt)

**Comment by boredland** [comment]
Hi there! I primarily chose open-meteo.com because they claim to be open-source and didn't require an API-Key that would somehow need to make its way to the end-user. But with our [proxy-setup](https://github.com/manjaro-sway/manjaro-sway/blob/main/docs/functions/weather/%5Blocation%5D.ts), we surely could use multiple different providers and still deduplicate requests.
